homeassistant/components/geniushub/__init__.py

```diff
@@ -105,20 +125,26 @@ def make_debug_log_entries(self):
 
 
 class GeniusEntity(Entity):
-    """Base for all Genius Hub endtities."""
+    """Base for all Genius Hub entities."""
 
-    def __init__(self):
+    def __init__(self, hub_uid: str):
         """Initialize the entity."""
-        self._name = None
+        self._hub_uid = hub_uid.upper()
```

Comment:
Any modification of the id should be done preferably in the config schema, or if needed in the broker.